sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -2117,7 +2265,9 @@ case class Cast(
     child: Expression,
     dataType: DataType,
     timeZoneId: Option[String] = None,
-    override val ansiEnabled: Boolean = SQLConf.get.ansiEnabled)
+    override val ansiEnabled: Boolean = SQLConf.get.ansiEnabled,
+    fallbackConfKey: String = SQLConf.ANSI_ENABLED.key,
+    fallbackConfValue: String = "false")
```

Comment:
`typeCheckFailureMessage` in `Cast` is different from `AnsiCast`. The former use `ANSI_ENABLED` but the latter use `STORE_ASSIGNMENT_POLICY`.

Reply:
Can we make it an abstract class so that implementations can override? I'm really worried about changing the class constructor as many spark plugins use `Cast.apply/unapply`.